crates/criticaltrust/src/v2/manifests.rs

```diff
@@ -0,0 +1,177 @@
+// SPDX-FileCopyrightText: The Ferrocene Developers
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! Serializable and deserializable representation of criticaltrust v2 manifests.
+
+use serde_semver::SemverReq;
+
+use crate::keys::KeyRole;
+use crate::signatures::Signable;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, PartialEq, Eq, SemverReq)]
+#[version("0.0.1")]
+pub struct MetadataVersion;
+
+impl std::fmt::Debug for MetadataVersion {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&format!("MetadataVersion: {}", MetadataVersion::version())[..])
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub struct Metadata {
+    pub release_identifier: String,
+    // TODO: channel should be an enum?
+    pub channel: String,
+    pub ferrocene_version: String,
+    pub rust_version: String,
+    pub release_date: String,
+    pub support_expires_date: String,
+    pub yanked: Option<Yanked>,
+    pub signature: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Copy)]
+pub enum MetadataKind {
+    #[serde(rename = "Ferrocene-Release")]
+    FerroceneRelease,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct Yanked {
+    pub code: u16,
+    pub human: String,
```

Comment:
It is unclear if this is "The human who yanked" or "A message for humans about why it was yanked", could we add a doc comment?